File: 3783-mirror-distance-of-an-integer/solution.py
class Solution:
    def mirrorDistance(self, n: int) -> int:
        """
        Calculates the mirror distance by converting the integer to a string,
        reversing it, and finding the absolute difference.
        """
        # Convert n to string, reverse the string, and convert back to integer
        # This naturally handles cases like n = 10 -> reverse = 01 -> 1
        reversed_n = int(str(n)[::-1])
        
        # Calculate and return the absolute difference
        return abs(n - reversed_n)

File: 3783-mirror-distance-of-an-integer/solution.cpp
#include <cmath>
#include <cstdlib>

class Solution {
public:
    /**
     * Calculates the absolute difference between n and its digit-wise reverse.
     * Uses long long to handle reversal safely.
     */
    long long mirrorDistance(int n) {
        long long original = n;
        long long reversedN = 0;
        long long temp = n;

        // Extract digits one by one and build the reversed number
        while (temp > 0) {
            int digit = temp % 10;
            reversedN = reversedN * 10 + digit;
            temp /= 10;
        }

        // Return the absolute difference
        return std::abs(original - reversedN);
    }
};